zrml/authorized/src/migrations.rs

```diff
@@ -0,0 +1,67 @@
+use crate::{AuthorizedOutcomeReports, Config, Outcomes, Pallet};
+use frame_support::{
+    dispatch::Weight,
+    log,
+    pallet_prelude::PhantomData,
+    traits::{Get, OnRuntimeUpgrade, StorageVersion},
+};
+
+const REQUIRED_STORAGE_VERSION: u16 = 0;
+const NEXT_STORAGE_VERSION: u16 = 1;
+
+pub struct MigrateAuthorizedStorage<T>(PhantomData<T>);
+
+// Due to problems with `storage_key_iter`, we temporarily use two maps. The old `Outcomes` will
+// no longer be used and removed at the next opportunity.
+impl<T: Config> OnRuntimeUpgrade for MigrateAuthorizedStorage<T> {
+    fn on_runtime_upgrade() -> Weight
+    where
+        T: Config,
+    {
+        if StorageVersion::get::<Pallet<T>>() != REQUIRED_STORAGE_VERSION {
+            return 0;
+        }
+        let mut total_weight: Weight = 0;
+        log::info!("Starting storage migration of authorized reports");
+
+        for (market_id, _, outcome_report) in <Outcomes<T>>::iter() {
```

Comment:
We want to empty the previous storage to avoid having a "zombie storage" (A storage that is not exposed anymore but contains data). The function [`drain`](https://docs.substrate.io/rustdocs/latest/frame_support/pallet_prelude/struct.StorageDoubleMap.html#method.drain) can be used for that.

Reply:
Fixed in 6745af9